### ISSUE TITLE
OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed

### ISSUE THREAD

**Issue by illtellyoulater** [claudiob/yt#227]
ruby -v 
`ruby 2.2.2p95 (2015-04-13 revision 50295) [x64-mingw32]`

gem -v
`2.4.8`

gem query yt

```
*** LOCAL GEMS ***

yt (0.25.4)
```

Full error:

```
irb(main):019:0> Yt.configure do |config|
irb(main):020:1*   config.api_key = '123123123123'
irb(main):021:1> end
=> "123123123123"
irb(main):022:0> videos = Yt::Collections::Videos.new
=> #<Yt::Collections::Videos:0x000000039f1610 @parent=nil, @auth=nil>
irb(main):023:0> videos.where(order: 'viewCount').first.title #=>  "PSY - GANGNAM STYLE"
Yt::Errors::ServerError: {"request_curl":"curl -X GET -H \"content-length: 0\" -H \"user-agent: Yt::Request (gzip)\" \"https://www.googleapis.com/youtube/v3/search?key=123123123123\u0026maxResults=50\u0026order=viewCount\u0026part=snippet\u0026type=video\"","response_body":"#\u003cOpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed\u003e"}
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/request.rb:83:in `run'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/request.rb:81:in `run'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/actions/list.rb:94:in `fetch_page'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/actions/list.rb:84:in `next_page'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/collections/videos.rb:78:in `next_page'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/actions/list.rb:50:in `find_next'
        from C:/Dev/Ruby22-x64/lib/ruby/gems/2.2.0/gems/yt-0.25.4/lib/yt/actions/list.rb:21:in `block in list'
        from C:in `each'
        from C:in `each'
        from C:in `first'
        from C:in `first'
        from (irb):23
        from C:/Dev/Ruby22-x64/bin/irb:11:in `<main>'
```

What can I do to fix it?
Thanks.

**Comment by claudiofullscreen** [comment]
@jjgh Can this help? https://github.com/Fullscreen/yt/issues/103

**Comment by illtellyoulater** [comment]
@claudiofullscreen no I think it doesn't, It's a different error, this gives no error:

```
irb(main):001:0>  require 'openssl'
=> true
irb(main):002:0> OpenSSL::SSL::SSLErrorWaitReadable
=> OpenSSL::SSL::SSLErrorWaitReadable
```

 :(

**Comment by claudiofullscreen** [comment]
Sorry @jjgh – I meant to link https://github.com/Fullscreen/yt/issues/163

**Comment by illtellyoulater** [comment]
Yep I got to the same solution through another path :) -> https://gist.github.com/fnichol/867550

Quoting @Befzz :

```
1. Download http://curl.haxx.se/ca/cacert.pem
2. Save it as %USERPROFILE%/cacert.pem ( C:\Users\__you__\cacert.pem)
3. Set persistent environment variable for current user SSL_CERT_FILE as %USERPROFILE%/cacert.pem
```

Now it works! Thanks!

**Comment by illtellyoulater** [comment]
Damnit! I spent the last 3 hours fighting with this!

I did not want to set a permanent environment variable in Windows, so I was doing:

`ENV['SSL_CERT_FILE'] = Dir.pwd + '/cacert.pem'`

by which I was simply trying to set a process-life time-spanning environment variable (pointing to "cacert.pem" in the app directory) just for my script.... but it was erroring out again as before, despite that debugging with `system 'set SSL_CERT_FILE'` (which would just print the value of the env var) was reporting the variable to be set correctly. The only clue I had was that in the following little script with just the minimum lines of code it was working!

```
ENV['SSL_CERT_FILE'] = Dir.pwd + '/cacert.pem' 
require 'yt'
videos = Yt::Collections::Videos.new
puts videos.where(order: 'viewCount').first.title
```

So after disabling almost everything step by step in my real script at the end I was baffled to find out that for some obscure reason I just needed to put `ENV['SSL_CERT_FILE'] = Dir.pwd + '/cacert.pem'` at the top of the file for it to work!

..and they just call em "gotchas".. phew! 
hope it helps someone else..
